packages/devextreme/js/core/utils/object.js

```diff
@@ -62,11 +75,15 @@ const deepExtendArraySafe = function(target, changes, extendComplexObject, assig
 
         if(isPlainObject(newValue)) {
             const goDeeper = extendComplexObject ? isObject(prevValue) : isPlainObject(prevValue);
-            newValue = deepExtendArraySafe(goDeeper ? prevValue : {}, newValue, extendComplexObject, assignByReference);
+            newValue = deepExtendArraySafe(goDeeper ? prevValue : {}, newValue, extendComplexObject, assignByReference, shouldCopyUndefined);
         }
 
-        if(newValue !== undefined && prevValue !== newValue) {
-            assignValueToProperty(target, name, newValue, assignByReference);
+        if(
+            Array.isArray(newValue) && !assignByReference ||
+            shouldCopyUndefined && (prevValue !== newValue || prevValue === undefined) ||
+            newValue !== undefined && prevValue !== newValue
```

Comment:
Let's move out this complex conditions outside the ```if``` statement into a few separate variables